spec/std/number_spec.cr

```diff
@@ -1,6 +1,48 @@
 require "spec"
 
+private macro it_initialize_from_value_to(number_type)
```

Comment:
Why is this a macro? Seems like a method would do.

Reply:
Minor grammar issue: `it_initializes_from_value_to` (ditto below)

Reply:
Oh. That's unexpected. Why is that...

Reply:
Hm. I think `be_a` doesn't need to be a macro.

Reply:
https://carc.in/#/r/5wnn

Reply:
@straight-shoota That's true and I tried to change it in the pass and then reverted it because it currently doesn't work for actual class types: https://carc.in/#/r/5woq

The reason is that the type of `String.class` is `Class` and matching `T.class` of `Class` gives `Object` because a `Class` is an instance of an Object... it's probably not correct, but there's no way to go from `Class` to the actual instance type.

The actual problem is that the type of `"foo"` is String, and matching that with `T.class` gives the `String` **class** (called metaclass in the compiler) type, whereas when matching against `T` it gives you the String **instance** type. Now the value of `String.class` is the String **class*, but it's type is `Class`. And `Class` is a generic placeholder for all classes. Instead, we should probably make it be an endless type creation, so it would be the "String class class", and matching it against `T.class` would give the "String class". It's a bit confusing, but I think in Objective-C it works like that. I'll try it out soon.